projects/gubernator/pipeline.yaml

```diff
@@ -0,0 +1,173 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: gubernator
+metadata:
+  name: gubernator
+  openshift.io/display-name: Gubernator
+  iconClass: icon-storage
+  tags: instant-app,storage
+  template.openshift.io/long-description: |-
+    The Gubernator interface is a web frontend for displaying CI
+    test results stored in GCS.
+parameters:
+- name: APP
+  displayName: Application Name
+  description: Name of the larger application this deployment is part of.
+  value: k8sci
+  required: false
+- name: COMPONENT
+  displayName: Application Component Name
+  description: Name of this component in the larger application
+  value: gubernator
+  required: false
+- name: NAME
+  displayName: Application Name
+  description: The name to use for the objects created by this template
+  value: gubernator
+  required: true
+- name: SOURCE_REF
+  displayName: Source Ref
+  description: Git reference to use for building the Gubernator image
+  value: master
+  required: false
+- name: SOURCE_URL
+  displayName: Source URL
+  description: Git source URL to use for building the Gubernator image
+  value: https://github.com/openshift/test-infra.git
+  required: true
+- name: SOURCE_CONTEXT
+  displayName: Source Context
+  description: Git source context to use for building the Gubernator image
+  value: gubernator
+  required: false
+- name: SERVICE_NAME
+  displayName: Gubernator Service Name
+  description: The name of the OpenShift Service exposed for the Gubernator container.
+  value: gubernator
+objects:
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+    labels:
+      app: "${APP}"
+      component: "${COMPONENT}"
+- kind: BuildConfig
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+    labels:
+      app: "${APP}"
+      component: "${COMPONENT}"
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: "${NAME}:latest"
+    source:
+      git:
+        ref: "${SOURCE_REF}"
+        uri: "${SOURCE_URL}"
+      contextDir: "${SOURCE_CONTEXT}"
+      type: Git
+      dockerfile: |
+        FROM gcr.io/google-appengine/python
+
+        RUN virtualenv /env
+        ENV VIRTUAL_ENV /env
+        ENV PATH /env/bin:$PATH
+
+        ADD requirements.txt /app/requirements.txt
+        RUN pip install -r /app/requirements.txt
+        ADD . /app
+        RUN ln -s /app/secrets.json /var/run/secrets/gubernator/secrets.json
```

Comment:
Could try to push this upstream... do they care to deploy on k8s? If so, if using a volume mount on their pod they will not be able to put it down at the same level as everything else as the mount will nuke the directory